storage/src/main/java/com/tencent/rss/storage/handler/impl/ShuffleIndexHeader.java

```diff
@@ -53,26 +65,65 @@ public int getHeaderLen() {
     return (int)ShuffleStorageUtils.getIndexFileHeaderLen(partitionNum);
   }
 
+  // No side effects on byteBuffer
+  public static ShuffleIndexHeader extractHeader(ByteBuffer byteBuffer) {
```

Comment:
make it as a construct?

Reply:
We need to handle the extraction failure and return null so it would be more complicated if using constructor either in handling exception or construct an empty ShuffleIndexHeader.